### PR TITLE
Add framework scheme

### PR DESCRIPTION
**Pull request by 8W9aG**
* Carthage now supported :dart:

### CODE REVIEW

**Comment by 8W9aG** [comment]
@rastersize @dflems @jgavris

File: .travis.yml [hunk 1]
```diff
@@ -20,6 +20,8 @@ env:
     - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=9.1,name=iPhone 5"
     - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=9.2,name=iPhone 5"
     - TEST_SDK=macosx TEST_DEST="platform=OS X"
+    - TEST_SDK=watchOS TEST_DEST="platform=watchOS"
```
Comment (rastersize) [inline .travis.yml]: Change to
```yaml
    - TEST_SDK=watchsimulator2.1 TEST_DEST="OS=2.1,name=Apple Watch - 38mm"
```

File: .travis.yml [hunk 2]
```diff
@@ -20,6 +20,8 @@ env:
...
+    - TEST_SDK=tvOS TEST_DEST="platform=tvOS"
```
Comment (rastersize) [inline .travis.yml]: Change to
```yaml
    - TEST_SDK= appletvsimulator9.1 TEST_DEST="OS=9.1,name=Apple TV 1080p"
```

**Comment by jgavris** [comment]
Configure new schemes for test action

**Comment by rastersize** [comment]
Think we’ll need to extend the build matrix with something like:
```yml
env:
  global:
    - LC_CTYPE=en_US.UTF-8
    - LANG=en_US.UTF-8
    - SCHEME_IOS_LIBRARY="SPTDataLoader"
    - SCHEME_IOS_FRAMEWORK="SPTDataLoader-iOS"
    - SCHEME_OSX_FRAMEWORK="SPTDataLoader-OSX"
    - SCHEME_TV_FRAMEWORK="SPTDataLoader-TV"
    - SCHEME_WATCH_FRAMEWORK="SPTDataLoader-Watch"
  matrix:
    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=8.1,name=iPhone 5" SCHEME="$SCHEME_IOS_LIBRARY"
    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=8.1,name=iPhone 5" SCHEME="$SCHEME_IOS_FRAMEWORK"
    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=8.2,name=iPhone 5" SCHEME="$SCHEME_IOS_FRAMEWORK"
    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=8.3,name=iPhone 5" SCHEME="$SCHEME_IOS_FRAMEWORK"
    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=8.4,name=iPhone 5" SCHEME="$SCHEME_IOS_FRAMEWORK"
    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=9.0,name=iPhone 5" SCHEME="$SCHEME_IOS_FRAMEWORK"
    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=9.0,name=iPhone 5" SCHEME="$SCHEME_IOS_LIBRARY"
    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=9.1,name=iPhone 5" SCHEME="$SCHEME_IOS_FRAMEWORK"
    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=9.2,name=iPhone 5" SCHEME="$SCHEME_IOS_FRAMEWORK"
    - TEST_SDK=macosx TEST_DEST="platform=OS X,arch=x86_64" SCHEME="$SCHEME_OSX_FRAMEWORK"
    - TEST_SDK=watchsimulator2.1 TEST_DEST="OS=2.1,name=Apple Watch - 38mm" SCHEME="$SCHEME_WATCH_FRAMEWORK"
    - TEST_SDK= appletvsimulator9.1 TEST_DEST="OS=9.1,name=Apple TV 1080p" SCHEME="$SCHEME_TV_FRAMEWORK"
```

**Comment by jgavris** [comment]
Looking good @rastersize. I would appreciate tab-spaced variables for readability of the matrix.

File: .travis.yml [hunk 3]
```diff
@@ -11,15 +11,27 @@ env:
...
+    - TEST_SDK=iphonesimulator  TEST_DEST="platform=iOS Simulator,OS=8.4,name=iPhone 5" SCHEME="$SCHEME_IOS_FRAMEWORK"
+    - TEST_SDK=iphonesimulator  TEST_DEST="platform=iOS Simulator,OS=9.0,name=iPhone 5" SCHEME="$SCHEME_IOS_FRAMEWORK"
+    - TEST_SDK=iphonesimulator  TEST_DEST="platform=iOS Simulator,OS=9.0,name=iPhone 5" SCHEME="$SCHEME_IOS_LIBRARY"
+    - TEST_SDK=iphonesimulator  TEST_DEST="platform=iOS Simulator,OS=9.1,name=iPhone 5" SCHEME="$SCHEME_IOS_FRAMEWORK"
+    - TEST_SDK=iphonesimulator  TEST_DEST="platform=iOS Simulator,OS=9.2,name=iPhone 5" SCHEME="$SCHEME_IOS_FRAMEWORK"
+    - TEST_SDK=macosx           TEST_DEST="platform=OS X,arch=x86_64"                   SCHEME="$SCHEME_OSX_FRAMEWORK"
+    - TEST_SDK=watchsimulator   TEST_DEST="OS=2.1,name=Apple Watch - 38mm"              SCHEME="$SCHEME_WATCH_FRAMEWORK"
+    - TEST_SDK=watchsimulator   TEST_DEST="OS=2.0,name=Apple Watch - 38mm"              SCHEME="$SCHEME_WATCH_FRAMEWORK"
+    - TEST_SDK=appletvsimulator TEST_DEST="OS=9.1,name=Apple TV 1080p"                  SCHEME="$SCHEME_TV_FRAMEWORK"
+    - TEST_SDK=appletvsimulator TEST_DEST="OS=9.0,name=Apple TV 1080p"                  SCHEME="$SCHEME_TV_FRAMEWORK"
```
Comment (jgavris) [inline .travis.yml]: :+1:

File: .travis.yml [hunk 4]
```diff
@@ -11,15 +11,27 @@ env:
...
   matrix:
-    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=8.1,name=iPhone 5"
-    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=8.2,name=iPhone 5"
-    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=8.3,name=iPhone 5"
-    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=8.4,name=iPhone 5"
-    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=9.0,name=iPhone 5"
-    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=9.1,name=iPhone 5"
-    - TEST_SDK=iphonesimulator TEST_DEST="platform=iOS Simulator,OS=9.2,name=iPhone 5"
-    - TEST_SDK=macosx TEST_DEST="platform=OS X"
+    - TEST_SDK=iphonesimulator  TEST_DEST="platform=iOS Simulator,OS=8.1,name=iPhone 5" SCHEME="$SCHEME_IOS_LIBRARY"            BUILD='test'
```
Comment (rastersize) [inline .travis.yml]: Nit: How about renaming `BUILD` to `ACTIONS` instead?

File: ci/script.sh [hunk 5]
```diff
@@ -2,10 +2,10 @@
 
 set -euo pipefail
 
-xcrun xcodebuild build test \
+xcrun xcodebuild build "$BUILD_ACTIONS" \
```
Comment (rastersize) [inline ci/script.sh]: Remove the extra `build` here.
```shell
xcrun xcodebuild "$BUILD_ACTIONS" \
```

**Comment by rastersize** [comment]
This is starting to look pretty good now I must say :sunglasses::+1:

**Pull request merged**